modules/keepnick.cpp

```diff
@@ -163,12 +163,20 @@ class CKeepNickMod : public CModule {
         return CONTINUE;
     }
 
-    EModRet OnRaw(CString& sLine) override {
-        // Are we trying to get our primary nick and we caused this error?
-        // :irc.server.net 433 mynick badnick :Nickname is already in use.
-        if (m_pTimer && sLine.Token(1) == "433" &&
-            sLine.Token(3).Equals(GetNick()))
-            return HALT;
+    EModRet OnNumericMessage(CNumericMessage& numeric) override {
+        //              bad nick                    chan is +N (unrealircd)     banned on chan (charybdis)
+        if (m_pTimer && (numeric.GetCode() == 433 || numeric.GetCode() == 447 || numeric.GetCode() == 435)) {
+            CString sLine = numeric.ToString();
```

Comment:
CNumericMessage already has the message parsed. Why do you need to stringify it, and then parse with Token() again?
Also, by doing this you're adding a bogus `:` in the beginning of error message.